### PR TITLE
*: integrate plugin framework with TiDB (#9006)

### PR DESCRIPTION
**Pull request by lysu**
cherry-pick #9006 to 2.1

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/9888)

### CODE REVIEW

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1 pd=release-2.1

File: executor/show.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package executor
 import (
 	"bytes"
 	"fmt"
+	"github.com/pingcap/tidb/plugin"
```
Comment (jackysp) [inline executor/show.go]: move it down to 3rd libs

**Comment by jackysp** [comment]
Should we upgrade the go version on release-2.1?

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by zz-jason** [comment]
LGTM

**Comment by lysu** [comment]
/integration-common-test  tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by jackysp** [comment]
/run-integration-common-test tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by jackysp** [comment]
/run-integration-common-test tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Pull request merged**